src/smc-webapp/frame-editors/latex-editor/actions.ts

```diff
@@ -268,17 +270,36 @@ export class Actions extends BaseActions<LatexEditorState> {
 
   async run_build(time: number, force: boolean): Promise<void> {
     this.setState({ build_logs: Map() });
+
+    // for knitr related documents, we have to first build the derived tex file ...
     if (this.knitr) {
       await this.run_knitr(time, force);
       if (this.store.get("knitr_error")) return;
     }
-    await this.run_latex(time, force);
+    // update_pdf=false, because it is defered until the end
+    await this.run_latex(time, force, false);
```

Comment:
the `false` here is new: it tells `run_latex` to not update the pdf preview, because there might be an additional sagetex or pythontex pass. below here, `update_pdf` is called.